index.js

```diff
@@ -53,71 +53,125 @@ function render_rules(vulnerabilities) {
-                                       },
-                                       "analysisTarget": {
-                                       "uri": dockerfile_location,
-                                       "index": 0
-                                       },
-                                       "locations": [
-                                       {
-                                           "physicalLocation": {
-                                           "artifactLocation": {
-                                               "uri": dockerfile_location
-                                           },
-                                           "region": {
-                                               "startLine": 1,
-                                               "startColumn": 1,
-                                               "endLine": 1,
-                                               "endColumn": 1,
-                                               "byteOffset": 1,
-                                               "byteLength": 1
-                                           }
-                                           },
-                                           "logicalLocations": [
-                                           {
-                                               "fullyQualifiedName": "dockerfile"
-                                           }
-                                           ]
-                                       }
-                                       ],
-                                       "suppressions": [
-                                       {
-                                           "kind": "external"
-                                       }
-                                       ],
-                                       "baselineState": "unchanged"
-                                   }
-                                   }
-                                  ) 
+function getLocation(v) {
+  if (v.artifact.locations.length) {
+    // If the scan was against a directory, the location will be a string
+    var location = v.artifact.locations[0];
+    if (typeof location === "string") {
+      return location;
     }
-    return(ret);
+    // Otherwise it is an object with "path" and "layer" keys
+    return location["path"];
+  }
+  // XXX there is room for improvement here, trying to mimick previous behavior
+  // If no `dockerfile-path` was provided, and in the improbable situation where there
+  // are no locations for the artifact, return 'Dockerfile'
+  return "Dockerfile";
+}
+
+function textMessage(v) {
+  var scheme = sourceScheme();
+  var prefix;
+  if (["dir", "tar"].includes(scheme)) {
+    prefix =
+      "The path " +
+      getLocation(v) +
+      " would result in an installed vulnerability: ";
+  } else {
+    prefix = "The container image contains software with a vulnerability: ";
+  }
+  return prefix + "(" + v.package + " type=" + v.package_type + ")";
```

Comment:
[Template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) would make this line of strings a bit less awkward to work with

Reply:
https://github.com/anchore/scan-action/issues/51